extensions/wikia/Chat2/ChatAjax.class.php

```diff
@@ -208,63 +230,93 @@ static public function getListOfBlockedPrivate() {
 	 * returns "error" => [error message].
 	 */
 	static public function giveChatMod() {
-		ChatHelper::info( __METHOD__ . ': Method called' );
 		global $wgRequest, $wgUser;
+
 		wfProfileIn( __METHOD__ );
+		Chat::info( __METHOD__ . ': Method called' );
+
+		if ( !self::authenticateServer() ) {
+			wfProfileOut( __METHOD__ );
+
+			return [ 'error' => self::ERROR_NOT_AUTHENTICATED ];
+		}
 
-		$promottingUser = $wgUser;
+		$promotingUser = $wgUser;
 
-		$retVal = array();
+		$res = [ ];
 		$PARAM_NAME = "userToPromote";
```

Comment:
Why do we even need this variable? In `blockOrBanChat` we do a similar check for the `userToBan` param, but it is just a bare string.

Would extracting this to a const make more sense?